modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/TestClassWrapper.java

```diff
@@ -228,33 +230,35 @@ public void runTestMethods(@NotNull ITestRunManagers managers, IDynamicStatusSto
             }     
         }
 
+        setResult(getResult(), managers);
+
         // Run @AfterClass methods
         for (GenericMethodWrapper afterClassMethod : afterClassMethods) {
             afterClassMethod.invoke(managers, this.testClassObject, null);
             if (afterClassMethod.fullStop()) {
-                if (this.result == null) {
-                    this.result = Result.failed("AfterClass method failed");
+                if (getResult() == null) {
+                    setResult( Result.failed("AfterClass method failed") , managers );
                 }
             }
         }
```

Comment:
Would it be worth splitting parts of this method out into smaller methods? The bits to run the `@BeforeClass` and `@AfterClass` are very similar, and the bit that runs the `@Test` methods could also be separated?

(I know this code was already there, just thinking about tidying it up!)

Reply:
Thanks Eamonn - I've now refactored this method and split the running of the BeforeClass/AfterClass methods and running of Test methods into their own methods.